File: my-folder/0014-longest-common-prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if (!strs.length) return "";

    let prefix = strs[0];

    for (let i = 0; i < strs[0].length; i++) {
        const letter = strs[0][i];

        for (let j = 1; j < strs.length; j++) {
            if (letter !== strs[j][i]) {
                prefix = prefix.slice(0, i);
            }
        }
    }

    return prefix;
};

File: my-folder/0014-longest-common-prefix/solution.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        if (strs is null) return "";

        string prefix = strs[0];

        for (int j = 1; j < strs.Length; j++) {
            while (!strs[j].StartsWith(prefix)) {
                prefix = prefix.Substring(0, prefix.Length - 1);

                if (prefix is null) return "";
            }
        }

        return prefix;
    }
}
